commons-vfs2/src/test/java/org/apache/commons/vfs2/impl/DefaultFileSystemManagerTest.java

```diff
@@ -47,4 +50,34 @@ public void test_close() throws FileSystemException {
         Assert.assertFalse(VFS.getManager().resolveFile(Paths.get("DoesNotExist.not").toUri()).exists());
     }
 
+    @Test
+    public void testAddAndRemoveProvider() throws FileSystemException {
+        try (DefaultFileSystemManager fileSystemManager = new DefaultFileSystemManager()) {
+            fileSystemManager.setFilesCache(new NullFilesCache());
+            fileSystemManager.setCacheStrategy(CacheStrategy.MANUAL);
+
+            // this is to check when exactly `close` of the provider is called
+            Boolean[] closed = { Boolean.FALSE };
```

Comment:
Hi @boris-petrov 
Thank you for your update.
This is a bit of an anti-pattern here, as in general you should use an `AtomicBoolean` instead of a 1-element array. 
BUT, since this is a test and you want to verify that a method was called, this is done using Mockito which we already depend on and you can simply use by following the pattern described here https://stackoverflow.com/questions/9841623/mockito-how-to-verify-method-was-called-on-an-object-created-within-a-method

Reply:
I changed the code to use `AtomicBoolean`. However, I'm unsure which pattern exactly you want me to use.

Reply:
Actually I did it with Mockito. Check it out.